streams/src/test/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessorMetricsTest.java

```diff
@@ -0,0 +1,229 @@
+        "stream-buffer-metrics",
+        "The current count of buffered records.",
+        mkMap(
+            mkEntry("client-id", "mock-processor-context-virtual-thread"),
+            mkEntry("task-id", "0_0"),
+            mkEntry("buffer-id", "test-store")
+        )
+    );
+
+    private static final MetricName BUFFER_COUNT_MAX_METRIC = new MetricName(
+        "suppression-buffer-count-max",
+        "stream-buffer-metrics",
+        "The max count of buffered records.",
+        mkMap(
+            mkEntry("client-id", "mock-processor-context-virtual-thread"),
+            mkEntry("task-id", "0_0"),
+            mkEntry("buffer-id", "test-store")
+        )
+    );
+
+    @Test
+    public void shouldRecordMetrics() {
+        final String storeName = "test-store";
+
+        final StateStore buffer = new InMemoryTimeOrderedKeyValueBuffer.Builder(storeName)
+            .withLoggingDisabled()
+            .build();
+
+        final KTableSuppressProcessor<String, Long> processor =
+            new KTableSuppressProcessor<>(
+                (SuppressedInternal<String>) Suppressed.<String>untilTimeLimit(Duration.ofDays(100), maxRecords(1)),
+                storeName,
+                String(),
+                new FullChangeSerde<>(Long())
+            );
+
+        final MockInternalProcessorContext context = new MockInternalProcessorContext();
+        context.setCurrentNode(new ProcessorNode("testNode"));
+
+        buffer.init(context, buffer);
+        processor.init(context);
+
+        final long timestamp = 100L;
+        context.setStreamTime(timestamp);
+        context.setRecordMetadata("", 0, 0L, null, timestamp);
+        final String key = "longKey";
+        final Change<Long> value = new Change<>(null, ARBITRARY_LONG);
+        processor.process(key, value);
+
+        {
+            final Map<MetricName, ? extends Metric> metrics = context.metrics().metrics();
+
+            verifyMetric(metrics, EVICTION_RATE_METRIC, is(0.0));
+            verifyMetric(metrics, EVICTION_TOTAL_METRIC, is(0.0));
+            verifyMetric(metrics, SUPPRESSION_RATE_METRIC, greaterThan(0.0));
+            verifyMetric(metrics, SUPPRESSION_TOTAL_METRIC, is(1.0));
+            verifyMetric(metrics, BUFFER_SIZE_AVG_METRIC, is(51.0));
+            verifyMetric(metrics, BUFFER_SIZE_CURRENT_METRIC, is(51.0));
+            verifyMetric(metrics, BUFFER_SIZE_MAX_METRIC, is(51.0));
+            verifyMetric(metrics, BUFFER_COUNT_AVG_METRIC, is(1.0));
+            verifyMetric(metrics, BUFFER_COUNT_CURRENT_METRIC, is(1.0));
+            verifyMetric(metrics, BUFFER_COUNT_MAX_METRIC, is(1.0));
+        }
+
+        context.setStreamTime(timestamp + 1);
+        context.setRecordMetadata("", 0, 1L, null, timestamp + 1);
+        processor.process("key", value);
+
+        {
+            final Map<MetricName, ? extends Metric> metrics = context.metrics().metrics();
+
+            verifyMetric(metrics, EVICTION_RATE_METRIC, greaterThan(0.0));
```

Comment:
Is this rate value be time-dependent (I remember we saw some issues with it if the test runs too slowly), to be non-zero in some edge cases?

Reply:
IIRC, rate metrics are time-dependent, but I think that this particular comparison is pretty safe for a mock processor unit test.